src/modules/sidebar/partials/SharingEntryLink.vue

```diff
@@ -0,0 +1,345 @@
+<!--
+  - SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<template>
+	<div class="sharing-entry-link">
+		<!-- Create Link State -->
+		<div v-if="isCreateMode">
+			<div v-if="!showCreateForm" class="sharing-entry-link__create">
+				<NcButton
+					type="secondary"
+					class="sharing-entry-link__create-button"
+					:disabled="loading"
+					@click="showCreateForm = true">
+					<template #icon>
+						<Plus :size="20" />
+					</template>
+					{{ t('tables', 'Create public link') }}
+				</NcButton>
+			</div>
+
+			<!-- Inline Creation Form -->
+			<div v-else class="sharing-entry-link__create-form">
+				<div class="sharing-entry-link__form-row">
+					<NcActionCheckbox
+						:checked.sync="usePassword"
+						@update:chk="onTogglePassword">
+						{{ t('tables', 'Password protection') }}
+					</NcActionCheckbox>
+				</div>
+
+				<div v-if="usePassword" class="sharing-entry-link__form-row">
+					<NcActionInput
+						:value.sync="password"
+						type="text"
```

Comment:
Should this be rather a type password? I think even in the files app that is not shown in plain text